modules/ding_base/ding_base.module

```diff
@@ -535,3 +535,38 @@ function ding_base_rewrite_color_stylesheet() {
   variable_set('color_' . $theme . '_stylesheets', $css);
   variable_set('color_' . $theme . '_files', $paths['files']);
 }
+
+/**
+ * Implements hook_file_presave().
+ */
+function ding_base_file_presave($file) {
+  // Check if uploaded file is image.
+  if (empty(file_validate_is_image($file))) {
+    $toolkits = image_get_available_toolkits();
+
+    // Check if gd2 library available.
+    if (array_key_exists('gd', $toolkits)) {
```

Comment:
We are using ImageMagick since a case in current sprint. Is GD2 always
available by default?

On Saturday, 13 February 2016, AM <notifications@github.com> wrote:

> In modules/ding_base/ding_base.module
> <https://github.com/easySuite/ding2/pull/23#discussion_r52831159>:
>
> > @@ -535,3 +535,38 @@ function ding_base_rewrite_color_stylesheet() {
> >    variable_set('color_' . $theme . '_stylesheets', $css);
> >    variable_set('color_' . $theme . '_files', $paths['files']);
> >  }
> > +
> > +/**
> > + * Implements hook_file_presave().
> > + */
> > +function ding_base_file_presave($file) {
> > +  // Check if uploaded file is image.
> > +  if (empty(file_validate_is_image($file))) {
> > +    $toolkits = image_get_available_toolkits();
> > +
> > +    // Check if gd2 library available.
> > +    if (array_key_exists('gd', $toolkits)) {
>
> Drupal on its own relies on gd. So this scenario is barely possible. 11
> февр. 2016 г. 12:57 пользователь "Martin Cording" <
> notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> написал:
> … <#-2106010891_>
> In modules/ding_base/ding_base.module <#23 (comment)
> <https://github.com/easySuite/ding2/pull/23#discussion_r52586211>>: > @@
> -535,3 +535,38 @@ function ding_base_rewrite_color_stylesheet() { >
> variable_set('color_' . $theme . '_stylesheets', $css); >
> variable_set('color_' . $theme . '_files', $paths['files']); > } > + > +/**
> > + * Implements hook_file_presave(). > + */ > +function
> ding_base_file_presave($file) { > + // Check if uploaded file is image. > +
> if (empty(file_validate_is_image($file))) { > + $toolkits =
> image_get_available_toolkits(); > + > + // Check if gd2 library available.
> > + if (array_key_exists('gd', $toolkits)) { What if GD2 is not available?
> — Reply to this email directly or view it on GitHub <
> https://github.com/easySuite/ding2/pull/23/files#r52586211>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/easySuite/ding2/pull/23/files#r52831159>.
>


-- 


Med venlig hilsen
Martin Cording | Udviklings- og leverancechef
Inlead
Rentemestervej 56C - 2400 København NV - Danmark
Tel.: +45 70 20 12 20 <+4570201220> - Mob: +45 20 71 79 86 <+4520717986>
www.inlead.dk | mc@inlead.dk | CVR 10158826